File: my-solutions/0215-kth-largest-element-in-an-array/solution.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        minheap = []

        for i in range(k):
            heapq.heappush(minheap, nums[i])

        for i in range(k, len(nums)):
            if nums[i] > minheap[0]:
                heapq.heappush(minheap, nums[i])
                heapq.heappop(minheap)
        
        return minheap[0]

File: my-solutions/0215-kth-largest-element-in-an-array/solution.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        nth_element(nums.begin(), nums.begin() + k - 1, nums.end(), greater<int>());
        // for(auto& num: nums) cout << num << " ";
        return nums[k - 1];
    }
};
